include/zephyr/drivers/misc/morse_code/morse_code.h

```diff
@@ -0,0 +1,93 @@
+/* morse_code.h - Public API for the Morse Code driver */
+/*
+ * Copyright (c) 2024 Freedom Veiculos Eletricos
+ * Copyright (c) 2024 OS Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_MORSE_CODE_H_
+#define ZEPHYR_INCLUDE_MORSE_CODE_H_
+
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Morse Code APIs
+ * @defgroup morse_code Morse code APIs
+ * @ingroup misc_interfaces
+ * @{
+ */
+
+/**
+ *  @brief Send data using morse code
+ *
+ * This translates the text in morse code and sends it over a GPIO pin. This
+ * function is non-blocking and will return immediately. The manage callback
+ * should be used to synchronize between FSM and process.
+ *
+ *  @param dev Pointer to device structure for driver instance.
+ *  @param data the ASCII text to send
+ *  @param size the length of the text in bytes. The 0 value question the driver
+ *  about processing queue.
+ *
+ * @return positive number to indicate missing bytes to transmit
+ * @return 0 when device is free
+ * @return -EINVAL when some argument is invalid
+ * @return -EBUSY when driver is transmitting
+ * @return -ENOENT when symbol to be transmitted is not a valid morse code
+ */
+int morse_code_send(const struct device *dev, const uint8_t *data,
+		    const size_t size);
```

Comment:
This is a good blocking API call - I could see it being made non-blocking as well.

Can we add a `k_timeout_t timeout` parameter to it? I guess the timeout would represent the max time to wait to start transmitting?

An alternative would be to specify a deadline for the whole message. If it's impossible to send the whole message when the "radio" becomes free, it could also be a kind of timeout.

```suggestion
int morse_code_send(const struct device *dev, const char *data,
		    const size_t size, k_timeout_t timeout);
```